src/handler/guild/channel.ts

```diff
@@ -0,0 +1,260 @@
+
+import { Guild as GuildEntity } from "../../entity/Guild";
+
+export class Channel {
+    @CommandData({ type: "guild" })
+    channelData() {
+        return new SlashCommandBuilder()
+            .setName("channel")
+            .setDescription("Configure channels")
+            .setDefaultPermission(false)
+            .addSubcommand((subcommand) =>
+                subcommand
+                    .setName("add")
+                    .setDescription("Add new channel")
+                    .addStringOption((option) =>
+                        option
+                            .setName("type")
+                            .setDescription(
+                                "Which type of channel to configure"
+                            )
+                            .setRequired(true)
+                            .addChoices([
+                                ["Log", "log"],
+                                ["Verify", "verify"],
+                                ["Troop Movements", "troop_movement"],
+                                ["Facility Updates", "land_facility"],
+                                ["Command", "command"],
+                            ])
+                    )
+                    .addChannelOption((option) =>
+                        option
+                            .setName("channel")
+                            .setDescription("Which channel to use")
+                            // broken typings :/
+                            /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+                            .addChannelType(ChannelType.GuildText as any)
+                            .setRequired(true)
+                    )
+            )
+            .addSubcommand((subcommand) =>
+                subcommand
+                    .setName("remove")
+                    .setDescription("Unset a configured channel")
+                    .addStringOption((option) =>
+                        option
+                            .setName("type")
+                            .setDescription("Which type of channel to unset")
+                            .setRequired(true)
+                            .addChoices([
+                                ["Log", "log"],
+                                ["Verify", "verify"],
+                                ["Troop Movements", "troop_movement"],
+                                ["Facility Updates", "land_facility"],
+                                ["Command", "command"],
+                            ])
+                    )
+                    .addChannelOption((option) =>
+                        option
+                            .setName("channel")
+                            .setDescription("Which channel to use")
+                            // broken typings :/
+                            /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+                            .addChannelType(ChannelType.GuildText as any)
```

Comment:
shouldnt this be required?

Reply:
no, not only for the command channels. Of the other channels there can (as of now) only be one per guild, so it's not necessary to specify which one you mean